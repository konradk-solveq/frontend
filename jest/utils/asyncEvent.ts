import {act} from '@testing-library/react-native';

const asyncEvent = async (event: any) => {
    await act(async () => {
        await event;
    });
};

export default asyncEvent;
