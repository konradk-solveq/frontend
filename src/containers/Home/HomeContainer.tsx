import React from 'react';
import {
    View,
    StyleSheet,
    GestureResponderEvent,
    Image,
    ImageSourcePropType,
    Pressable,
    ScrollView,
} from 'react-native';
import {Header1} from '@components/texts/texts';
import {useMergedTranslation} from '@utils/translations/useMergedTranslation';
import {getFVerticalPx, getFHorizontalPx} from '@helpers/appLayoutDimensions';
import {AddBikeTile} from '@components/tiles';
import colors from '@theme/colors';
import {appContainerHorizontalMargin} from '@theme/commonStyle';

interface IProps {
    userName?: string;
    onAddBikePressPrimary: (e: GestureResponderEvent) => void;
    onAddBikePressSecondary: (e: GestureResponderEvent) => void;
    storeImageSource?: ImageSourcePropType;
    onStoreTilePress?: (e: GestureResponderEvent) => void;
}

const HomeContainer = ({
    userName,
    onAddBikePressPrimary,
    onAddBikePressSecondary,
    onStoreTilePress,
}: IProps) => {
    const {t} = useMergedTranslation('MainHome');
    return (
        <View style={styles.container}>
            <ScrollView>
                <Header1 style={styles.header}>
                    {`${t('hi')}`}
                    {userName ? ` ${userName}` : ''}!
                </Header1>
                <AddBikeTile
                    onPressPrimary={onAddBikePressPrimary}
                    onPressSecondary={onAddBikePressSecondary}
                />
                <Pressable style={styles.storeTile} onPress={onStoreTilePress}>
                    <Image
                        source={require('@assets/images/homeStore/storeTile.png')}
                        resizeMode={'contain'}
                        style={styles.image}
                    />
                </Pressable>
            </ScrollView>
        </View>
    );
};

export default HomeContainer;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        width: '100%',
        backgroundColor: colors.whiteGrey,
        paddingHorizontal: appContainerHorizontalMargin,
    },
    header: {
        textAlign: 'left',
        width: '100%',
        marginVertical: getFVerticalPx(24),
    },
    storeTile: {
        height: getFVerticalPx(180),
        width: getFHorizontalPx(358),
        borderRadius: getFHorizontalPx(12),
    },
    image: {
        height: '100%',
        width: '100%',
    },
});
