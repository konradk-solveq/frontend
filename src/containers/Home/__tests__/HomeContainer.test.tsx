import 'react-native';
import React from 'react';
import {fireEvent} from '@testing-library/react-native';

import HomeContainer from '../HomeContainer';
import renderComponent from '@jestUtils/render';
import asyncEvent from '@jestUtils/asyncEvent';

const TEST_USER_NAME = 'User';
const BANNER_LINK = 'home-container-banner-link';

describe('<HomeContainer /> - containers/home/HomeContainer', () => {
    const onAddBikePressPrimary = jest.fn();
    const onAddBikePressSecondary = jest.fn();
    const onStoreTilePress = jest.fn();

    it('Should match snapshot', async () => {
        const component = await asyncEvent(
            renderComponent(
                <HomeContainer
                    userName={TEST_USER_NAME}
                    onAddBikePressPrimary={onAddBikePressPrimary}
                    onAddBikePressSecondary={onAddBikePressSecondary}
                    onStoreTilePress={onStoreTilePress}
                />,
            ),
        );

        expect(component).toMatchSnapshot();
    });

    it('Should access to banner button', async () => {
        const {getByTestId} = await asyncEvent(
            renderComponent(
                <HomeContainer
                    userName={TEST_USER_NAME}
                    onAddBikePressPrimary={onAddBikePressPrimary}
                    onAddBikePressSecondary={onAddBikePressSecondary}
                    onStoreTilePress={onStoreTilePress}
                />,
            ),
        );

        const bannerLink = getByTestId(BANNER_LINK);
        await asyncEvent(fireEvent.press(bannerLink));
        expect(onStoreTilePress).toBeCalled();
    });

    afterEach(() => {
        jest.restoreAllMocks();
        jest.clearAllMocks();
    });
});
