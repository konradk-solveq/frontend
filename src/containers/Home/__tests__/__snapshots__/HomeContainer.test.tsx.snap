// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeContainer /> - containers/home/HomeContainer Should match snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "flex": 1,
        "paddingHorizontal": 30.76923076923077,
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "paddingBottom": 202.3127962085308,
        }
      }
      showsVerticalScrollIndicator={false}
    >
      <View>
        <Text
          adjustsFontSizeToFit={false}
          numberOfLines={0}
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "DIN2014-Demi",
                "fontSize": 24,
                "fontWeight": "600",
                "lineHeight": 32,
                "textAlign": "left",
              },
              Object {
                "marginVertical": 37.93364928909953,
                "textAlign": "left",
                "width": "100%",
              },
            ]
          }
          testID="header1-test-id"
        >
          MainHome.welcomeHeader
           User
          !
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 18.966824644549764,
                "marginBottom": 25.28909952606635,
                "paddingVertical": 37.93364928909953,
              },
              undefined,
            ]
          }
          testID="add-bike-tile"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 165.95971563981044,
                  "justifyContent": "center",
                  "marginBottom": 12.644549763033176,
                  "width": 165.95971563981044,
                },
                undefined,
              ]
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={165.95971563981044}
              bbWidth={165.95971563981044}
              fill="none"
              focusable={false}
              height={165.95971563981044}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 165,
                    "width": 165,
                  },
                ]
              }
              vbHeight={512}
              vbWidth={512}
              width={165.95971563981044}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="m162.631 110.372-15.79 4.863 16.838 54.695 14.731-8.304-15.779-51.254ZM99.097 342.709c13.682 0 24.774-11.092 24.774-24.774s-11.092-24.774-24.774-24.774-24.774 11.092-24.774 24.774 11.092 24.774 24.774 24.774Z"
                  fill={4292532443}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M412.903 218.839c-54.73 0-99.097 44.367-99.097 99.097 0 54.729 44.367 99.097 99.097 99.097S512 372.665 512 317.935c0-54.729-44.367-99.096-99.097-99.096Zm0 173.419c-41.047 0-74.323-33.275-74.323-74.323 0-41.048 33.275-74.323 74.323-74.323 41.048 0 74.323 33.275 74.323 74.323 0 41.048-33.276 74.323-74.323 74.323Z"
                  fill={4284306272}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M412.903 235.355c-45.608 0-82.581 36.972-82.581 82.581 0 45.608 36.973 82.581 82.581 82.581 45.608 0 82.581-36.973 82.581-82.581 0-45.609-36.973-82.581-82.581-82.581Zm0 148.645c-36.486 0-66.065-29.578-66.065-66.065 0-36.487 29.578-66.065 66.065-66.065 36.487 0 66.065 29.578 66.065 66.065 0 36.487-29.578 66.065-66.065 66.065Z"
                  fill={4293716716}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M99.097 218.839C44.367 218.839 0 263.206 0 317.935c0 54.729 44.367 99.097 99.097 99.097s99.097-44.367 99.097-99.097c0-54.729-44.367-99.096-99.097-99.096Zm0 173.419c-41.047 0-74.323-33.275-74.323-74.323 0-41.048 33.276-74.323 74.323-74.323 41.047 0 74.323 33.275 74.323 74.323-.001 41.048-33.276 74.323-74.323 74.323Z"
                  fill={4284306272}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M99.097 235.355c-45.608 0-82.58 36.972-82.58 82.581 0 45.608 36.972 82.581 82.58 82.581s82.581-36.973 82.581-82.581c-.001-45.609-36.973-82.581-82.58-82.581Zm0 148.645c-36.486 0-66.065-29.578-66.065-66.065 0-36.487 29.578-66.065 66.065-66.065 36.487 0 66.065 29.578 66.065 66.065C165.161 354.422 135.583 384 99.097 384Z"
                  fill={4293716716}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M90.84 317.935h16.515v33.032H90.839v-33.032Z"
                  fill={4290163127}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M52.992 290.419a8.221 8.221 0 0 1-5.073-1.746c-3.597-2.802-4.234-7.992-1.435-11.589 1.694-2.169 3.492-4.286 5.484-6.278s4.113-3.79 6.274-5.48c3.597-2.815 8.79-2.165 11.597 1.435 2.798 3.597 2.16 8.786-1.435 11.589-1.645 1.278-3.25 2.625-4.758 4.133s-2.855 3.117-4.13 4.754a8.252 8.252 0 0 1-6.524 3.182ZM134.879 372.29a8.254 8.254 0 0 1-6.524-3.181c-2.798-3.597-2.161-8.786 1.435-11.589 1.645-1.278 3.25-2.625 4.758-4.133s2.855-3.117 4.129-4.754c2.806-3.597 7.992-4.254 11.597-1.435 3.597 2.802 4.234 7.992 1.435 11.589-1.694 2.169-3.492 4.286-5.484 6.278s-4.113 3.79-6.274 5.48a8.215 8.215 0 0 1-5.072 1.745ZM459.008 290.419a8.254 8.254 0 0 1-6.524-3.181c-1.274-1.637-2.621-3.246-4.129-4.754-1.508-1.508-3.113-2.855-4.758-4.133-3.597-2.802-4.234-7.992-1.435-11.589 2.798-3.601 8-4.25 11.597-1.435 2.161 1.69 4.282 3.488 6.274 5.48 1.992 1.992 3.79 4.109 5.484 6.278 2.798 3.597 2.161 8.786-1.435 11.589a8.225 8.225 0 0 1-5.074 1.745ZM377.121 372.29a8.221 8.221 0 0 1-5.073-1.746c-2.161-1.69-4.282-3.488-6.274-5.48-1.992-1.992-3.79-4.109-5.484-6.278-2.798-3.597-2.161-8.786 1.435-11.589 3.589-2.819 8.79-2.161 11.597 1.435 1.274 1.637 2.621 3.246 4.129 4.754 1.508 1.508 3.113 2.855 4.758 4.133 3.597 2.802 4.234 7.992 1.435 11.589a8.246 8.246 0 0 1-6.523 3.182Z"
                  fill={4294951529}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="m419.355 321.036-21.484-26.855a90.865 90.865 0 0 1-15.697-29.408l-32.761-103.742-10.487-7.914-6.827 7.914H177.66a16.517 16.517 0 0 0-14.808 9.201L91.694 314.27l6.315 11.851 123.871 16.516c.349.046.697.068 1.046.07.015 0 .028.007.043.007.798 0 1.621-.121 2.427-.371.149-.046.25-.161.395-.214.504-.184.933-.477 1.394-.757.503-.306.996-.59 1.413-.988.116-.11.276-.154.387-.271l119.55-127.023 17.895 56.674a107.325 107.325 0 0 0 18.538 34.731l21.484 26.855a8.253 8.253 0 0 0 6.46 3.101 8.238 8.238 0 0 0 5.153-1.81c3.556-2.847 4.137-8.044 1.29-11.605Zm-307.742-9.762 58.943-118.991 40.724 132.28-99.667-13.289Zm115.072 7.175-43.378-140.9h154.006l5.523 17.491-116.151 123.409Z"
                  fill={4294934662}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="m363.355 128-8.255-8.258h-13.511a16.574 16.574 0 0 0-13.323 6.758c-3.097 4.23-4.008 9.742-2.419 14.734l6.251 19.798h17.315l-7.824-24.774h13.508l8.258-8.258Z"
                  fill={4292532443}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M123.871 104.044c0 12.394 11.338 21.689 23.491 19.258l51.794-10.359a9.076 9.076 0 0 0-1.78-17.976h-64.429a9.077 9.077 0 0 0-9.076 9.077ZM222.968 359.226c13.682 0 24.774-11.092 24.774-24.774s-11.092-24.774-24.774-24.774-24.774 11.092-24.774 24.774 11.092 24.774 24.774 24.774Z"
                  fill={4284306272}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M230.839 331.948c-1.387-4.343-6.016-6.738-10.379-5.367-4.339 1.383-6.742 6.028-5.363 10.375l18.395 57.806 15.742-5.008-18.395-57.806Z"
                  fill={4283121486}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M256 400.516h-24.774a8.256 8.256 0 0 1-8.258-8.258 8.256 8.256 0 0 1 8.258-8.258H256a8.256 8.256 0 0 1 8.258 8.258 8.256 8.256 0 0 1-8.258 8.258ZM99.097 359.226a8.258 8.258 0 0 0 8.258-8.258 8.258 8.258 0 1 0-8.258 8.258ZM99.097 326.193a8.258 8.258 0 0 0 8.258-8.258 8.258 8.258 0 1 0-8.258 8.258Z"
                  fill={4284306272}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGRect
                  fill={4284306272}
                  height={28.450236966824647}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rx="9"
                  width={52.15876777251185}
                  x="351"
                  y="119"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginHorizontal": 23.076923076923077,
                },
                Object {
                  "paddingBottom": 37.93364928909953,
                },
              ]
            }
          >
            <Text
              adjustsFontSizeToFit={false}
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "DIN2014-Demi",
                    "fontSize": 20,
                    "fontWeight": "600",
                    "lineHeight": 24,
                    "textAlign": "center",
                  },
                  Object {
                    "paddingBottom": 12.644549763033176,
                  },
                ]
              }
              testID="header2-test-id"
            >
              MainBike.AddBikeTile.header
            </Text>
            <Text
              adjustsFontSizeToFit={false}
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "color": "#717171",
                    "fontFamily": "DIN2014-Demi",
                    "fontSize": 18,
                    "fontWeight": "600",
                    "lineHeight": 24,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
              testID="header3-test-id"
            >
              MainBike.AddBikeTile.body
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "paddingHorizontal": 61.53846153846154,
                },
              ]
            }
          >
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 30.76923076923077,
                    "flexDirection": "row",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  },
                  Object {},
                  Array [
                    Object {
                      "height": 92.3076923076923,
                      "width": 565.3846153846154,
                    },
                    Object {
                      "marginBottom": 25.28909952606635,
                    },
                  ],
                  Object {
                    "backgroundColor": "#C63733",
                  },
                ]
              }
              testID="add-bike-tile-primary-button"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 30.76923076923077,
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "center",
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
                testID="add-bike-tile-primary-button-container"
              >
                <Text
                  adjustsFontSizeToFit={true}
                  style={
                    Array [
                      Object {
                        "color": "#333333",
                        "fontFamily": "DIN2014-Demi",
                        "fontSize": 18,
                        "lineHeight": 28,
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                  testID="add-bike-tile-primary-button-text"
                >
                  MainBike.AddBikeTile.primaryButton
                </Text>
              </View>
            </View>
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 30.76923076923077,
                    "flexDirection": "row",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  },
                  Object {},
                  Array [
                    Object {
                      "height": 92.3076923076923,
                      "width": 565.3846153846154,
                    },
                    Object {},
                  ],
                  Object {
                    "backgroundColor": "#FFFFFF",
                  },
                ]
              }
              testID="add-bike-tile-secondary-button"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 30.76923076923077,
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "center",
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
                testID="add-bike-tile-secondary-button-container"
              >
                <Text
                  adjustsFontSizeToFit={true}
                  style={
                    Array [
                      Object {
                        "color": "#333333",
                        "fontFamily": "DIN2014-Demi",
                        "fontSize": 18,
                        "lineHeight": 28,
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#C63733",
                      },
                    ]
                  }
                  testID="add-bike-tile-secondary-button-text"
                >
                  MainBike.AddBikeTile.secondaryButton
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="home-container-banner-link"
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../src/assets/images/homeStore/storeTile.png",
              }
            }
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>,
  <View
    style={
      Array [
        Object {
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "backgroundColor": "#313131",
          "bottom": 0,
          "height": 0,
          "justifyContent": "flex-end",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "100%",
          "zIndex": 1000,
        }
      }
    >
      <Text
        collapsable={false}
        nativeID="animatedComponent"
        numberOfLines={1}
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "DIN2014Narrow-Regular",
            "fontSize": 13,
            "letterSpacing": 0.4,
            "opacity": 0,
            "paddingBottom": 7.444196428571429,
            "paddingHorizontal": 36.231884057971016,
            "textAlign": "center",
          }
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;
